Add rpms from install-deps.sh to Dockerfile.build-radosgw

tserong opened the pull request:
Currently, the build-radosgw container has a minimal set of packages necessary for the build, then later when you *run* the container, it calls out to `install-deps.sh` from the ceph source tree to install ceph's build dependencies (300+ rpms). This takes about 15 minutes (for me, at least, with my rural Tasmanian internet connection), and happens every time I run the build-radosgw container to build ceph.  If we put all these dependencies inside Dockerfile.build-radosgw instead, they only get downloaded once, when building that container.  We still run `install-deps.sh` in case any new dependencies have been added to ceph in the meantime, but at least we'll have almost all of them upfront.

~I've also added mirrorsorcerer which updates the zypper repos to whichever mirror is closest to you when you build the container.~

Signed-off-by: Tim Serong <tserong@suse.com>

jecluis reviewed (approved):
the really long line drives me nuts, but lgtm nonetheless. Thanks :)

votdev commented on build/Dockerfile.build-radosgw:
```diff
@@ -1,6 +1,13 @@
 FROM opensuse/tumbleweed
 
+# Find the fastest mirrors
+RUN zypper -n install mirrorsorcerer && mirrorsorcerer -x && zypper ref
```
This line is causing the following error:
```
 ~/git/s3gw-core/build   testing ✔  $ podman build --tag build-radosgw -f ./Dockerfile.build-radosgw
STEP 1/8: FROM opensuse/tumbleweed
STEP 2/8: RUN zypper -n install mirrorsorcerer && mirrorsorcerer -x && zypper ref
Retrieving repository 'openSUSE-Tumbleweed-Non-Oss' metadata [...done]
Building repository 'openSUSE-Tumbleweed-Non-Oss' cache [....done]
Retrieving repository 'openSUSE-Tumbleweed-Oss' metadata [.......done]
Building repository 'openSUSE-Tumbleweed-Oss' cache [....done]
Retrieving repository 'openSUSE-Tumbleweed-Update' metadata [.done]
Building repository 'openSUSE-Tumbleweed-Update' cache [....done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following NEW package is going to be installed:
  mirrorsorcerer

1 new package to install.
Overall download size: 1.0 MiB. Already cached: 0 B. After the operation, additional 3.4 MiB will be used.
Continue? [y/n/v/...? shows all options] (y): y
Retrieving package mirrorsorcerer-0.1.0~13-1.1.x86_64 (1/1),   1.0 MiB (  3.4 MiB unpacked)
Retrieving: mirrorsorcerer-0.1.0~13-1.1.x86_64.rpm [.done (5.5 MiB/s)]

Checking for file conflicts: [...done]
(1/1) Installing: mirrorsorcerer-0.1.0~13-1.1.x86_64 [............done]
 INFO ✨ Mirror Sorcerer ✨ 
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
 WARN Error creating pinger e=IOError(Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" })
ERROR Mirror profiling failed!
Error: error building at STEP "RUN zypper -n install mirrorsorcerer && mirrorsorcerer -x && zypper ref": error while running runtime: exit status 1
```

jecluis replied:
This is likely a problem with podman running rootless, and the container trying to ping the outside.

Try this: https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md#enable-unprivileged-ping

votdev replied:
OK. But according to the docs it should work. My UID/GID is 1000.
```
$ sysctl net.ipv4.ping_group_range
net.ipv4.ping_group_range = 0	2147483647
```
Any idea?

votdev commented:
I saw the following output:
```
-- ssl soname: libssl.so.1.1
-- crypto soname: libcrypto.so.1.1
-- BUILDING Boost Libraries at j 14
-- boost will be downloaded...
-- Found nasm: best -- capable of assembling AVX512
-- mgr module disabled for 3.8.13: diskprediction_local
bash: line 1: jq: command not found
-- Found the following ICU libraries:
--   uc (required)
CMake Warning at src/rgw/store/dbstore/CMakeLists.txt:46 (message):
  Gtest not enabled
```
I think it makes sense to add `jq` to the package list as well, even if it is not really required to get a successful build.

tserong commented:
Cool, i'll add `jq`.

I'm totally fine to drop the mirrorsorcerer thing if it causes issues, especially since installing all this stuff upfront only happens once.

tserong commented:
I fixed the long line @jecluis, added `jq` (which _is_ listed as a BuildRequires in ceph.spec, but only when building `--with make_check` - given the error, possibly that should just be always installed, but that's fix for upstream). I also dropped mirrorsorcerer. My most recent test build of this container only took just over 8 minutes, so I guess my connection speed is a bit variable ;-)
